src/mlmc/quantity.py

```diff
@@ -112,31 +112,34 @@ def size(self) -> int:
         """
         return self.qtype.size()
 
-    def get_cache_key(self, chunk_spec):
+    def get_cache_key(self, level_id=None, chunk_id=None, chunk_slice=None, n_samples=None):
```

Comment:
Is `n_samples` necessary? The samples chunk is uniquely given by previous parameters (proposed as a single data class Chunk.

Reply:
We want to get several samples (`n_samples`) to determine the results data type.

`chunks_quantity_level = [q.samples(level_id=0, chunk_id=0, n_samples=10) for q in quantities]`